src/chainstate/coordinator/comm.rs

```diff
@@ -0,0 +1,243 @@
+use std::{
+    thread, process
+};
+use std::time::{
+    Duration, Instant
+};
+use std::sync::{
+    Arc, RwLock,
+    atomic::{Ordering, AtomicU64}
+};
+
+use crossbeam_channel::{bounded, Sender, Receiver, TrySendError};
+
+
+/// Trait for use by the ChainsCoordinator
+/// 
+pub trait CoordinatorNotices {
+    fn notify_stacks_block_processed(&mut self);
+    fn notify_sortition_processed(&mut self);
+}
+
+pub struct ArcCounterCoordinatorNotices {
+    pub stacks_blocks_processed: Arc<AtomicU64>,
+    pub sortitions_processed: Arc<AtomicU64>
+}
+
+impl CoordinatorNotices for () {
+    fn notify_stacks_block_processed(&mut self) {}
+    fn notify_sortition_processed(&mut self) {}
+}
+
+impl CoordinatorNotices for ArcCounterCoordinatorNotices {
+    fn notify_stacks_block_processed(&mut self) {
+        self.stacks_blocks_processed.fetch_add(1, Ordering::SeqCst);
+    }
+    fn notify_sortition_processed(&mut self) {
+        self.sortitions_processed.fetch_add(1, Ordering::SeqCst);
+    }
+}
+
+/// Structure used for communication _with_ a running
+///   ChainsCoordinator
+#[derive(Clone)]
+pub struct CoordinatorChannels {
+    // ChainsCoordinator takes two kinds of signals:
+    //    new stacks block & new burn block
+    // These signals can be coalesced -- the coordinator doesn't need
+    //    handles _all_ new blocks whenever it processes an event
+    //    because of this, we can avoid trying to set large bounds on these
+    //    event channels by using a coalescing thread.
+    new_stacks_block_channel: Sender<()>,
+    new_burn_block_channel: Sender<()>,
```

Comment:
I think I'm missing something here.  You mentioned in the PR that you needed the crossbeam crate in order to `select()` on multiple channels.  But, I'm having a hard time understanding why multiple channels even need to exist.  Can't there be a single channel that simply takes different message type variants -- i.e. one for announcing a new Stacks block, one for announcing a new burn block, and one for stopping the node?

Reply:
We don't want one type of event to crowd out the other type -- the only way to eliminate the possibility of crowding out entirely is by separating the event channels. Because the events are all coalescable, using three separate notice streams allow us to _never_ have the possibility of event dropping, while placing a very low bound on the memory requirements of the channels.

Reply:
Ah, that would make sense as to why it's not a single event channel.  But, if each event stream is made of coalescable events, then why use a channel abstraction at all?  Why not just have three mutex-guarded booleans, and a condition variable for waking up the coordinator thread when at least one of the booleans changes to `true`?

Reply:
If we do this -- switch to mutex-guarded booleans with a condition variable -- we can drop the crossbeam dependency altogether.

Reply:
Updated in 5b4cd3e